### PR TITLE
Prevent shared conf files from failing with different available options in different binaries

### PR DESCRIPTION
**Pull request by achow101**
From IRC:

```
<ossifrage> FYI, bitcoin-qt from the head I built today won't start if you have "daemon=0" in the config file, so you can't use the same config for either bitcoind or bitcoin-qt
<ossifrage> Seems like bitcoin-qt should ignore this option?
<provoostenator> ossifrage: probably caused by 13112. Another problem is disablewallet=1 will prevent a launch if you compile bitcoind without wallet. It probably needs to be relaxed slightly.
```

Adds all of the options that are unavailable due to compiling options to the hidden category so that shared config files do not break with the alternative binaries.

### CODE REVIEW

**Comment by ken2812221** [comment]
utACK 009f737.
Maybe also take qt options into consideration.

**Comment by achow101** [comment]
> Maybe also take qt options into consideration.

Oh right. Added those.

**Review by kallewoof — COMMENTED** [review]
utACK f6b81b23102377e9c4e81c65d1faf54306bd0c73

All these lines makes me want to have a `gArgs.AddDisabledArgs("-addresstype", "-changetype", ...)` method though (could take a vector or vargs).

**Comment by achow101** [comment]
> All these lines makes me want to have a `gArgs.AddDisabledArgs("-addresstype", "-changetype", ...)` method though (could take a vector or vargs).

I have added such a method which takes a vector of strings to add as hidden arguments.

File: src/init.cpp [hunk 1]
```diff
@@ -433,6 +454,11 @@ void SetupServerArgs()
     gArgs.AddArg("-zmqpubhashtx=<address>", "Enable publish hash transaction in <address>", false, OptionsCategory::ZMQ);
     gArgs.AddArg("-zmqpubrawblock=<address>", "Enable publish raw block in <address>", false, OptionsCategory::ZMQ);
     gArgs.AddArg("-zmqpubrawtx=<address>", "Enable publish raw transaction in <address>", false, OptionsCategory::ZMQ);
+#else
+    hidden_args.emplace_back("-zmqpubhashblock=<address>");
+    hidden_args.emplace_backg("-zmqpubhashtx=<address>");
```
[the remark this answers is not in the document]
Comment (achow101) [reply src/init.cpp]: Fixed

**Review by kallewoof — APPROVED** [review]
utACK c2dfbb4a97513557fe923b7810ea8639c320fefd

**Review by ken2812221 — APPROVED** [review]
utACK c2dfbb4

**Comment by MarcoFalke** [comment]
utACK c2dfbb4a97513557fe923b7810ea8639c320fefd

**Pull request merged**